### PR TITLE
Cleanup superthis

### PR DESCRIPTION
**Pull request by mabels**
Again

### CODE REVIEW

File: src/blockstore/connection-base.ts [hunk 1]
```diff
@@ -41,28 +53,28 @@ export abstract class ConnectionBase implements Connection {
     await (await throwFalsy(this.loader).WALStore()).process();
   }
 
-  async connect_X({ loader }: { readonly loader?: Loadable }) {
-    if (!loader) throw this.logger.Error().Msg("loader is required").AsError();
-    await this.connectMeta_X({ loader });
-    await this.connectStorage_X({ loader });
+  async connect(refl: RefLoadable | RefBlockstore) {
```
Comment (jchris) [inline src/blockstore/connection-base.ts:56]: thanks for the cleanup

File: src/blockstore/connection-base.ts [hunk 2]
```diff
@@ -41,28 +53,28 @@ export abstract class ConnectionBase implements Connection {
...
     if (rgateway.isErr())
       throw this.logger.Error().Result("err", rgateway).Url(metaUrl).Msg("connectMeta_X: gateway is required").AsError();
-    const name = metaUrl.toString();
-    const dbName = metaUrl.getParam("name");
-    if (!dbName) throw this.logger.Error().Url(metaUrl).Msg("connectMeta_X: dbName is required").AsError();
-
+    // const name = metaUrl.toString();
+    const dbName = metaUrl.getParam(PARAM.NAME);
+    if (!dbName) {
+      throw this.logger.Error().Url(metaUrl).Msg("connectMeta_X: dbName is required").AsError();
```
Comment (jchris) [inline src/blockstore/connection-base.ts]: connectMeta_X message is outdated

File: src/blockstore/gateway.ts [hunk 3]
```diff
@@ -17,17 +19,17 @@ export type UnsubscribeResult = Result<() => void>;
 export interface Gateway {
   // all the methods never throw!
   // an error is reported as a Result
-  buildUrl(baseUrl: URI, key: string): Promise<Result<URI>>;
+  buildUrl(baseUrl: URI, key: string, loader?: Loadable): Promise<Result<URI>>;
```
Comment (jchris) [inline src/blockstore/gateway.ts:22]: why is loader not set in the constructor?

Comment (jchris) [reply src/blockstore/gateway.ts:22]: is it b/c they wrap each other at call time?

File: src/database.ts [hunk 4]
```diff
@@ -70,7 +63,7 @@ export interface Database<DT extends DocTypes = NonNullable<unknown>> extends Ha
   readonly sthis: SuperThis;
   readonly id: string;
 
-  name(): string;
+  readonly name: string;
```
Comment (jchris) [inline src/database.ts:66]: thanks

File: src/runtime/gateways/file/gateway.ts [hunk 5]
```diff
@@ -75,21 +77,30 @@ export class FileGateway implements Gateway {
...
-  async put(url: URI, body: Uint8Array): Promise<Result<void>> {
+  async put<T>(url: URI, env: FPEnvelope<T>): Promise<Result<void>> {
+    // const rbuf = FPMsgMatch2Envelope(body) as Result<FPEnvelopeCar | FPEnvelopeFile>;
+    // if (rbuf.isErr()) {
+    //   return Result.Err(rbuf.Err());
+    // }
+    // let payload = body
+    // if (["car", "file"].includes(rbuf.Ok().type)) {
+    //   payload = rbuf.Ok().payload
+    // }
```
Comment (jchris) [inline src/runtime/gateways/file/gateway.ts]: remove ?

File: tests/blockstore/interceptor-gateway.test.ts [hunk 6]
```diff
@@ -75,51 +76,54 @@ describe("InterceptorGateway", () => {
...
-      "destroy",
-      "destroy",
-      "destroy",
-      "destroy",
-      "destroy",
-      "destroy",
-      "destroy",
-      "destroy",
-    ]);
+    expect(gwi.fn.mock.calls.map((i) => i[0]).sort() /* not ok there are some operation */).toEqual(
```
Comment (jchris) [inline tests/blockstore/interceptor-gateway.test.ts:82]: what is broken?

File: tests/www/todo.html [hunk 7]
```diff
@@ -164,7 +164,7 @@
           const timeout =
             10 +
             Math.pow(db._crdt.clock.head.length + cx.loader.taskManager.queue.length, 2) *
-              (Math.floor(Math.random() * 2000) + 2000);
+              (Math.floor(Math.random() * 1000) + 1000);
```
Comment (jchris) [inline tests/www/todo.html:167]: 🏃

**Review by jchris — APPROVED** [review]
there are comments worth addressing but technically ok with a merge

File: tests/blockstore/interceptor-gateway.test.ts (hunk 6 above)
Comment (mabels) [reply tests/blockstore/interceptor-gateway.test.ts:82]: There are some async thing where close has to wait for.

File: src/blockstore/gateway.ts (hunk 3 above)
Comment (mabels) [reply src/blockstore/gateway.ts:22]: this is in the middle of something 
there is not state in gateway we have just one for the hole system.

Comment (mabels) [reply src/blockstore/gateway.ts:22]: and as you see it's optional it's build for the key attachment thing

File: src/blockstore/gateway.ts [hunk 8]
```diff
@@ -23,13 +23,19 @@ export interface Gateway {
...
   delete(url: URI, loader?: Loadable): Promise<VoidResult>;
+
   // be notified of remote meta
   subscribe?(url: URI, callback: (meta: FPEnvelopeMeta) => Promise<void>, loader?: Loadable): Promise<UnsubscribeResult>;
+
+  // this method is used to get the raw data mostly for testing
+  getPlain(url: URI, key: string, loader?: Loadable): Promise<Result<Uint8Array>>;
+  // this method is used for testing only
+  // to avoid any drama it should only enabled in test mode
+  destroy(baseUrl: URI, loader?: Loadable): Promise<VoidResult>;
```
Comment (jchris) [inline src/blockstore/gateway.ts:38]: nice

File: src/blockstore/types.ts [hunk 9]
```diff
@@ -280,7 +280,7 @@ export interface RefBlockstore {
 export interface Connection {
   // readonly loader?: Loadable;
   readonly loaded: Promise<void>;
-  connectMeta(ref: RefLoadable | RefBlockstore): void;
+  // connectMeta(ref: RefLoadable | RefBlockstore): void;
```
Comment (jchris) [inline src/blockstore/types.ts:278]: where did it go? *note to self*

**Review by jchris — APPROVED** [review]

**Comment by jchris** [comment]
only question is how you connect just meta or just storage as a remote

**Pull request merged**